File: backend/src/main/kotlin/org/icpclive/events/codeforces/api/data/CFRanklistRow.java
package org.icpclive.events.codeforces.api.data;

import java.util.List;

/**
 * @author egor@egork.net
 */
public class CFRanklistRow {
    public CFParty party;
    public int rank;
    public double points;
    public int penalty;
    public int successfulHackCount;
    public int unsuccessfulHackCount;
    public List<CFProblemResult> problemResults;
    public Long lastSubmissionTimeSeconds;
}

File: src/cds/src/main/kotlin/org/icpclive/cds/codeforces/api/data/CFRankListRow.kt
package org.icpclive.cds.codeforces.api.data

import kotlinx.serialization.Serializable

@Serializable
data class CFRankListRow(
    val party: CFParty,
    val rank: Int,
    val points: Double,
    val penalty: Int,
    val successfulHackCount: Int,
    val unsuccessfulHackCount: Int,
    val problemResults: List<CFProblemResult>,
    val lastSubmissionTimeSeconds: Long? = null,
)